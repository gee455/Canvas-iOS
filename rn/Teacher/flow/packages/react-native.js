//
// Copyright (C) 2016-present Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, version 3 of the License.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.
//

// @flow

declare var console: typeof console & { disableYellowBox: boolean }

export type PanResponderGestureState = {
  stateID: number,
  moveX: number,
  moveY: number,
  x0: number,
  y0: number,
  dx: number,
  dy: number,
  vx: number,
  vy: number,
}

export type KeyboardEventData = {
  endCoordinates: {
    width: number,
    height: number,
    screenX: number,
    screenY: number,
  },
}
