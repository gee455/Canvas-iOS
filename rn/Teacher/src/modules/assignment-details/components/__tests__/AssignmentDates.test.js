//
// Copyright (C) 2016-present Instructure, Inc.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//

/**
 * @flow
 */

import 'react-native'
import React from 'react'
import AssignmentDates from '../AssignmentDates'
import renderer from 'react-test-renderer'

const template = {
  ...require('../../../../__templates__/assignments'),
}

test('render with multiple due dates', () => {
  const assignment = template.assignment({
    due_at: undefined,
    all_dates: [template.assignmentDueDate({ base: true }), template.assignmentDueDate({ base: false })],
  })
  let tree = renderer.create(
    <AssignmentDates assignment={assignment} />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

test('render with multiple due dates', () => {
  const assignment = template.assignment({
    lock_at: (new Date(0).toISOString()),
  })
  let tree = renderer.create(
    <AssignmentDates assignment={assignment} />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
