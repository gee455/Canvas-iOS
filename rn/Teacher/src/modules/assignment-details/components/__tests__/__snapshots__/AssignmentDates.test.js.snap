// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render with multiple due dates 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "fontFamily": ".SFUIDisplay",
        },
        Object {
          "color": "#2D3B45",
          "fontSize": 16,
        },
        undefined,
        Object {
          "fontFamily": ".SFUIDisplay",
        },
      ]
    }
  >
    Multiple Due Dates
  </Text>
</View>
`;

exports[`render with multiple due dates 2`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    accessibilityLabel={undefined}
    accessible={true}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontFamily": ".SFUIDisplay",
          },
          Object {
            "color": "#2D3B45",
            "fontSize": 16,
          },
          Object {
            "paddingTop": 4,
          },
          Object {
            "fontFamily": ".SFUIDisplay",
          },
        ]
      }
      testID="assignment-details.assignment-dates.due-date-lbl"
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontFamily": ".SFUIDisplay",
            },
            Object {
              "color": "#2D3B45",
              "fontSize": 16,
            },
            Object {
              "fontWeight": "600",
            },
            Object {
              "fontFamily": ".SFUIDisplay-semibold",
            },
          ]
        }
      >
        Due:
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontFamily": ".SFUIDisplay",
            },
            Object {
              "color": "#2D3B45",
              "fontSize": 16,
            },
            Object {},
            Object {
              "fontFamily": ".SFUIDisplay",
            },
          ]
        }
      >
         May 31, 2037 at 11:59 PM
      </Text>
    </Text>
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "fontFamily": ".SFUIDisplay",
        },
        Object {
          "color": "#2D3B45",
          "fontSize": 16,
        },
        Object {
          "paddingTop": 4,
        },
        Object {
          "fontFamily": ".SFUIDisplay",
        },
      ]
    }
    testID="assignment-details.assignment-dates.due-for-lbl"
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontFamily": ".SFUIDisplay",
          },
          Object {
            "color": "#2D3B45",
            "fontSize": 16,
          },
          Object {
            "fontWeight": "600",
          },
          Object {
            "fontFamily": ".SFUIDisplay-semibold",
          },
        ]
      }
    >
      For:
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontFamily": ".SFUIDisplay",
          },
          Object {
            "color": "#2D3B45",
            "fontSize": 16,
          },
          Object {},
          Object {
            "fontFamily": ".SFUIDisplay",
          },
        ]
      }
    >
       -
    </Text>
  </Text>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontFamily": ".SFUIDisplay",
          },
          Object {
            "color": "#2D3B45",
            "fontSize": 16,
          },
          Object {
            "paddingTop": 4,
          },
          Object {
            "fontFamily": ".SFUIDisplay",
          },
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontFamily": ".SFUIDisplay",
            },
            Object {
              "color": "#2D3B45",
              "fontSize": 16,
            },
            Object {
              "fontWeight": "600",
            },
            Object {
              "fontFamily": ".SFUIDisplay-semibold",
            },
          ]
        }
      >
        Availability:
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontFamily": ".SFUIDisplay",
            },
            Object {
              "color": "#2D3B45",
              "fontSize": 16,
            },
            Object {},
            Object {
              "fontFamily": ".SFUIDisplay",
            },
          ]
        }
      >
         Closed
      </Text>
    </Text>
  </View>
</View>
`;
